scratch/copy-mk.py

```diff
@@ -127,24 +131,20 @@ async def async_main(args) -> int:
         "digitiser-address": mcast_groups[0].split("+")[0],
         "sync-time": sync_time,
         "band": band,
+        "develop": args.develop,
     }
 
     out_cmd = [os.path.join(os.path.dirname(__file__), "sim_correlator.py")]
-
-    for k, v in out_kwargs.items():
-        out_cmd.append(f"--{k}={v}")
-
-    if args.develop is not None:
-        out_cmd.append(f"--develop={args.develop}")
+    out_cmd += [f"--{k}={v}" for k, v in out_kwargs.items() if v is not None]
 
     if args.image_override is not None:
         for override in args.image_override:
             out_cmd.append(f"--image-override={override}")
 
     out_cmd.append(args.controller)
 
-    print(f"Executing: {out_cmd}", "\n")
-    os.execv(out_cmd[0], out_cmd[1:])
+    print(f"Executing: {out_cmd}\n")
+    os.execv(out_cmd[0], out_cmd)
```

Comment:
> - Fix removal of the first argument, which prevented the subarray
  product from inheriting the name from the subordinate it duplicates

Is this what you are referring to here?

Reply:
Ah, I see in the docs 
> In either case, the arguments to the child process should start with the name of the command being run, but this is not enforced.

I hadn't noticed that previously.

is `os.execv()` passing the args via a shell like bash, or is this the list like one would expect in a C context, `argv[0]`?

Reply:
execv doesn't involve a shell. out_cmd[0] becomes argv[0] in the invoked process.

Reply:
That does make sense.